distsql: panic in stream_decoder.go

```
make testlogic TESTFLAGS='-verbosity=10 -show-sql -test.v' FILES=aggregate
[...]
SELECT 3 FROM kv HAVING TRUE;
panic: no typing info received yet

goroutine 2947 [running]:
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*StreamDecoder).Types(0xc42233a380, 0x1, 0x0, 0x0)
	/home/arjun/goworkspace/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/stream_decoder.go:180 +0x147
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.processProducerMessage(0x27acfc0, 0xc421db84e0, 0x27c28c0, 0xc420449820, 0x2790080, 0xc421593c80, 0xc42233a380, 0xc4214816a0, 0xc420989260, 0xc421db8420, ...)
	/home/arjun/goworkspace/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/inbound.go:171 +0x425
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.processInboundStreamHelper.func2(0xc421f3a000, 0x27c28c0, 0xc420449820, 0xc4202ffec0, 0xc420989200, 0x27acfc0, 0xc421db84e0, 0x2790080, 0xc421593c80, 0xc42233a380, ...)
	/home/arjun/goworkspace/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/inbound.go:110 +0x11a
created by github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.processInboundStreamHelper
	/home/arjun/goworkspace/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/inbound.go:92 +0x1b8
Makefile:864: recipe for target 'testbaselogic' failed
make: *** [testbaselogic] Error 2
```

--- a/pkg/sql/distsqlrun/data.pb.go
+++ b/pkg/sql/distsqlrun/data.pb.go
@@ -507,7 +507,8 @@ type ProducerMessage struct {
 	Header *ProducerHeader `protobuf:"bytes,1,opt,name=header" json:"header,omitempty"`
 	// Typing information. There will be one DatumInfo for each element in a row.
 	// This field has to be populated on, or before, a ProducerMessage with data
-	// in it, and can only be populated once.
+	// in it, and can only be populated once. It can be nil if only zero length
+	// rows will be sent.
 	// TODO(andrei): It'd be nice if the typing information for streams would be
 	// configured statically at plan creation time, instead of being discovered
 	// dynamically through the first rows that flow.

--- a/pkg/sql/distsqlrun/data.proto
+++ b/pkg/sql/distsqlrun/data.proto
@@ -223,7 +223,8 @@ message ProducerMessage {
 
   // Typing information. There will be one DatumInfo for each element in a row.
   // This field has to be populated on, or before, a ProducerMessage with data
-  // in it, and can only be populated once.
+  // in it, and can only be populated once. It can be nil if only zero length
+  // rows will be sent.
   // TODO(andrei): It'd be nice if the typing information for streams would be
   // configured statically at plan creation time, instead of being discovered
   // dynamically through the first rows that flow.

--- a/pkg/sql/distsqlrun/stream_decoder.go
+++ b/pkg/sql/distsqlrun/stream_decoder.go
@@ -176,9 +176,6 @@ func (sd *StreamDecoder) GetRow(
 // Types returns the types of the columns; can only be used after we received at
 // least one row.
 func (sd *StreamDecoder) Types() []sqlbase.ColumnType {
-	if !sd.typingReceived {
-		panic("no typing info received yet")
-	}
 	types := make([]sqlbase.ColumnType, len(sd.typing))
 	for i := range types {
 		types[i] = sd.typing[i].Type

